### PR TITLE
Add configurable hourly and daily limit parameters to Javabuilder

### PR DESCRIPTION
**Pull request by bencodeorg**
Allows setting a limit per hour and limit per day parameter for Javabuilder, and passes that value to the authorizers as an environment variable (HTTP is the only one that actually needs it, but easier to pass to both WebSocket and HTTP). Values can be overridden by parameters passed in via the deploy script. Values are 10 per hour and 40 per day be default.

This PR does not actually implement any limits -- it just allows us to set the limit values. Part of preventing overuse of javabuilder work, details here:
- [Dev doc](https://docs.google.com/document/d/1A4dIzg6ZKRxZb3bcTSlEMU7YXDzgr21HN7dI9jIe0XM/edit#)
- [Jira item](https://codedotorg.atlassian.net/browse/JAVA-458)

## Testing story

I tested deploying without setting any explicit values in the deploy script call and saw environment variables being set for the HTTP authorizer appropriately. I deployed again withe explicit values as arguments, and saw those values reflected in the HTTP authorizer's environment variables.

### CODE REVIEW

File: template.yml.erb
Comment (jmkulwik) [inline template.yml.erb:416]: NIT: Intentional to add these to both lambdas? (Not sure that there's a straightforward way to omit this from just the WebSocket Lambda, so there may not be anything to do, but it may be worth adding a comment in case these get split out in the future)

**Comment by jmkulwik** [comment]
Mind linking the task and dev doc in the PR description?

**Review by jmkulwik — APPROVED** [review]
LGTM! Once this is merged, mind updating the "Deploying Javabuilder" doc to include the environment variables for the production deploy so the limits are set to their higher values in production?

File: template.yml.erb
Comment (bencodeorg) [reply template.yml.erb:416]: Yeah there might? be a way to add just to http, but think it's probably more confusing than it'd be worth. I'll add a comment.

**Comment by bencodeorg** [comment]
> LGTM! Once this is merged, mind updating the "Deploying Javabuilder" doc to include the environment variables for the production deploy so the limits are set to their higher values in production?

@jmkulwik just confirming, the daily limit we want to set is 300? See it mentioned in the dev doc but doesn't seem set in stone?

**Review by sanchitmalhotra126 — COMMENTED** [review]
For my understanding - does this change actually enforce the limits, or is this just a way for us to pass these parameters to the lambdas?

**Comment by bencodeorg** [comment]
> For my understanding - does this change actually enforce the limits, or is this just a way for us to pass these parameters to the lambdas?

@sanchitmalhotra126 it just allows us to pass the values, I've updated the PR description to clarify.

**Comment by sanchitmalhotra126** [comment]
> > For my understanding - does this change actually enforce the limits, or is this just a way for us to pass these parameters to the lambdas?
> 
> @sanchitmalhotra126 it just allows us to pass the values, I've updated the PR description to clarify.

Thanks for clarifying!

**Pull request merged**
